### PR TITLE
Add authority into benches

### LINKED ISSUE #1230: `unreachable` WASM instruction in pallet benchmarks
### Problem

Panic occurs, see detailed log

### Steps

1. Compile node with `runtime-benchmarks` feature
2. `./target/release/gear-node benchmark pallet --pallet=pallet_gear  --chain=dev   --extrinsic=*  --execution=wasm --wasm-execution=compiled --heap-pages=4096`

### Possible Solution

_No response_

### Notes

In-action failed build: https://github.com/gear-tech/gear/runs/7482975668?check_suite_focus=true

### Relevant Log Output

<details><summary>Click to expand/collapse</summary>
<p>

Error executing and verifying runtime benchmark: Execution aborted due to trap: wasm trap: wasm `unreachable` instruction executed
WASM backtrace:

    0: 0x1eee31 - <unknown>!rust_begin_unwind
    1: 0x52d7 - <unknown>!core::panicking::panic_fmt::h2c84c084997fbcc5
    2: 0x189eb6 - <unknown>!core::panicking::unreachable_display::h06ecc46aaa1847ed
    3: 0x197a42 - <unknown>!pallet_gear::internal::<impl pallet_gear::pallet::Pallet<T>>::spend_gas::h6934ac6aa86be5dd
    4: 0x1bdf40 - <unknown>!gear_core_processor::handler::handle_journal::h17fd4e4a8f4ea64d
    5: 0x189d4a - <unknown>!pallet_gear::pallet::Pallet<T>::process_queue::hfd25313c57f887ee
    6: 0xc7c62 - <unknown>!core::ops::function::FnOnce::call_once{{vtable.shim}}::hf02928506f6a489e
    7: 0x1855a8 - <unknown>!pallet_gear::benchmarking::<impl frame_benchmarking::utils::Benchmarking for pallet_gear::pallet::Pallet<T>>::run_benchmark::h95fe60e9e0066f55
    8: 0x150296 - <unknown>!<gear_runtime::Runtime as frame_benchmarking::utils::runtime_decl_for_Benchmark::Benchmark<sp_runtime::generic::block::Block<sp_runtime::generic::header::Header<u32,sp_runtime::traits::BlakeTwo256>,sp_runtime::generic::unchecked_extrinsic::UncheckedExtrinsic<sp_runtime::multiaddress::MultiAddress<<<sp_runtime::MultiSignature as sp_runtime::traits::Verify>::Signer as sp_runtime::traits::IdentifyAccount>::AccountId,()>,gear_runtime::Call,sp_runtime::MultiSignature,(frame_system::extensions::check_non_zero_sender::CheckNonZeroSender<gear_runtime::Runtime>,frame_system::extensions::check_spec_version::CheckSpecVersion<gear_runtime::Runtime>,frame_system::extensions::check_tx_version::CheckTxVersion<gear_runtime::Runtime>,frame_system::extensions::check_genesis::CheckGenesis<gear_runtime::Runtime>,frame_system::extensions::check_mortality::CheckMortality<gear_runtime::Runtime>,frame_system::extensions::check_nonce::CheckNonce<gear_runtime::Runtime>,frame_system::extensions::check_weight::CheckWeight<gear_runtime::Runtime>,pallet_transaction_payment::ChargeTransactionPayment<gear_runtime::Runtime>)>>>>::dispatch_benchmark::h1eaf70616015350b
    9: 0x12969c - <unknown>!Benchmark_dispatch_benchmark

</p>
</details>

### PR DESCRIPTION
**Pull request by breathx**
Resolves #1230.

- Authority and proper block initialisation added into benchmarks.
- Journal processing removed from benchmarks.
- Simple no-op run of benchmarks added to CI.

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -245,6 +245,11 @@ jobs:
       - name: "Test: Runtime upgrade and queue processing"
         run: ./scripts/gear.sh test runtime-upgrade
 
+      - name: "Test: Runtime benchmarks work"
+        run: |
+          cargo build -p gear-node --release --features runtime-benchmarks
```
Comment (ark0f) [inline .github/workflows/CI.yaml:250]: I'd add the feature to `./gear.sh build --features runtime-benchmarks` to avoid rebuild of deps

Comment (breathx) [reply .github/workflows/CI.yaml:250]: not all our workspace members have such feature: pallets + node only 
it also a bit changes logic of execution: adds new extrinsics and increases recursion limit
so I vote for rebuild for this case atm

**Review by ukint-vs — APPROVED** [review]

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (ark0f) [reply .github/workflows/CI.yaml:250]: > not all our workspace members have such feature: pallets + node only

Which is ok to cargo

> it also a bit changes logic of execution: adds new extrinsics and increases recursion limit

So what's the execution logic change?

Comment (ark0f) [reply .github/workflows/CI.yaml:250]: I see we do something similar with `pallet-gear` and your change adds only 3 minutes, so it's ok

**Review by ark0f — APPROVED** [review]

**Pull request merged**